src/main/java/frc/robot/subsystems/amp/Amp.java

```diff
@@ -0,0 +1,157 @@
+package frc.robot.subsystems.amp;
+
+import static edu.wpi.first.units.Units.*;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.math.geometry.Translation3d;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import edu.wpi.first.wpilibj2.command.sysid.SysIdRoutine;
+import frc.robot.Constants;
+import java.util.Optional;
+import java.util.function.DoubleSupplier;
+import org.littletonrobotics.junction.Logger;
+
+// Subsystem base automatically runs periodic, too scared to put stuff in robot.java
+public class Amp extends SubsystemBase {
+
+  private final AmpIOInputsAutoLogged inputs = new AmpIOInputsAutoLogged();
+  private Mechanism2d ampBarPivot;
+  private MechanismRoot2d root;
+  private MechanismLigament2d rotatingLigament;
+
+  private Pose3d sim3dPose;
+  private Translation3d zeroedTranslation3d;
+  private Pose3d zeroedPose3d;
+
+  SysIdRoutine sysId;
+  AmpIO io;
+
+  public Amp(AmpIO io) {
+    this.io = io;
+    // TODO: Tune feedback controllers, these values are MADE UP
+    switch (Constants.currentMode) {
+      case REAL:
+        io.configurePID(1.0, 0, 0);
+        break;
+      case REPLAY:
+        io.configurePID(1.0, 0.0, 0.0);
+        break;
+      case SIM:
+        io.configurePID(3.0, 0.0, 0.1);
+        break;
+      default:
+        break;
+    }
+
+    setUp2dMechSim();
+    // Using SysID is wild
+    sysId =
```

Comment:
if you want sys id for this subsystem (proly not needed bc we're not actually doing anything in sim with autos/siys id stuff) you should have separate sys id things for the pivot and spinner